### ISSUE TITLE
Bug: [prefer-readonly] should report all statically analyzable properties and property access

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10655]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.3&fileType=.ts&code=MYGwhgzhAEAqCmEAuBGaBvAUNaB6X0EAFgPYCuIAJtAHYlLQBO8ADiY0ttC4wJYBuYJPG6MSLaAF5oAdgDcmLgDMSJABQBKDFxxIivCAG0A5D3HGAulOgoADApwBfTM8yhIMBMgBM2nPkJSCmpmNg4uHgEhEUMAZQBPAFsAIxIQADpeYUYhditpeRdFdyg4RCQAZj88AmJyKlp6JlZ2ThxIwWFoUzEWY2tC5VVNat19CHSzCWk7B2hnVxLPcoAWaoC64OawpAB%2BCL5OmJR82QVnIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Test1 {
  // should not report
  private prop = 7;

  foo() {
    this['prop'] = 10;
  }
}

class Test2 {
  // should report
  private [Symbol.iterator] = 7;
}

class Test3 {
  // should not report
  private 'prop' = 7;

  foo() {
    this.prop = 10;
  }
}

class Test4 {
  // should report?
  private [1] = 7;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-readonly": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the rule not to report the first example and to report on the second example.

### Actual Result

The rule reports on the first example, with the auto-fixer creating a type error: `Cannot assign to 'prop' because it is a read-only property.`. The second example should be reported but is missed.

### Additional Info

I think this is very similar to #10211, and I noticed this while working on #10552. I think these can be tackled with [`getStaticMemberAccessValue`](https://github.com/typescript-eslint/typescript-eslint/blob/ea6fbeab024f0c7936a88c6726a921ec3c299c94/packages/eslint-plugin/src/util/misc.ts#L290-L310).

**Comment by bradzacher** [comment]
The reason  we don't support a lot of these cases is just cos it's really rare that you define a property with quotes that doesn't need quotes, or use computed property access if you don't need it. With these sorts of cases we often find that "the juice isn't worth the squeeze" so we leave them as a later exercise for when someone asks (which is often "never" 😅).

Happy to accept a PR for it though
